src/UI/DateInput.elm

```diff
@@ -0,0 +1,42 @@
+module UI.DateInput exposing (DateInput, PeriodComparison, PeriodDate, RangeDate, fromPosix, toDD_MM_YYYY)
+
+import Time
+import UI.Internal.DateInput as Internal
+
+
+{-| `DateInput.DateInput` is used for describing dates in the UI.
+-}
+type alias DateInput =
+    Internal.DateInput
+
+
+{-| `DateInput.PeriodComparison` is used to compare date and time ranges.
+-}
+type alias PeriodComparison =
+    Internal.PeriodComparison
+
+
+{-| `DateInput.PeriodDate` is used to contain both date and comparison information.
+-}
+type alias PeriodDate =
+    Internal.PeriodDate
```

Comment:
Alias to a record is not an opaque type, you can use `{ date : DateInput, comparison : PeriodComparison }` here.